whois-api/src/main/java/net/ripe/db/whois/api/httpserver/JettyBootstrap.java

```diff
@@ -288,11 +301,10 @@ protected TrustManagerFactory getTrustManagerFactoryInstance() {
         sslContextFactory.setKeyStorePassword(whoisKeystore.getPassword());
         sslContextFactory.setCipherComparator(HTTP2Cipher.COMPARATOR);
 
-        // enable optional client certificates
-        sslContextFactory.setWantClientAuth(true);
+        sslContextFactory.setNeedClientAuth(isClientCertificate);
```

Comment:
Don't do this for both secure connectors, only for the one with the client cert auth, i.e.
```
if (isClientCertificate) {
        // accept self-signed client certificates for authentication
        sslContextFactory.setNeedClientAuth(true);
        sslContextFactory.setValidateCerts(false);
        sslContextFactory.setTrustAll(true);
}
```

Reply:
I wanted to leave the previous code as it is.
the only difference between the previous connector and the one with client certificate is the setNeedClientAuth

Reply:
As discussed these 3 lines go together, only do this for the client auth connector

Reply:
updated